tests/test_api/test_endpoints.py

```diff
@@ -1092,6 +1152,219 @@ def test_it_traverses_multiple_stages_of_workflow_executions(
+        another DataObject as output. In addition, the NucleotideSequencing process is also
+        linked to the MetagenomeAnnotation workflow via the `was_informed_by` slot.
+
+        ```mermaid
+        graph
+            biosample --> |has_input| nucleotide_sequencing_process
+            nucleotide_sequencing_process --> |has_output| data_object
+            data_object --> |has_input| metagenome_annotation_workflow
+            metagenome_annotation_workflow --> |has_output| data_object
+            metagenome_annotation_workflow --> |was_informed_by| nucleotide_sequencing_process
+        ```
+        """
+        faker = Faker()
+
+        # Create a NucleotideSequencing DataGeneration that produces raw data
+        nucleotide_sequencing_b_id = "nmdc:ntseq-00-000002"
+        data_object_b = faker.generate_data_objects(
+            quantity=1, id="nmdc:dobj-00-000006"
+        )[0]
+        data_object_b["data_category"] = "instrument_data"
+
+        nucleotide_sequencing_b = faker.generate_nucleotide_sequencings(
+            quantity=1,
+            id=nucleotide_sequencing_b_id,
+            has_input=[self.biosample_id],
+            has_output=[data_object_b["id"]],
+            associated_studies=[self.study_id],
+        )[0]
+
+        # Create processed data object
+        data_object_b = faker.generate_data_objects(
+            quantity=1, id="nmdc:dobj-00-000007"
+        )[0]
+        data_object_b["data_category"] = "processed_data"
+
+        # Create MetagenomeAnnotation workflow informed by the NucleotideSequencing
+        metagenome_annotation_b = faker.generate_metagenome_annotations(
+            quantity=1,
+            has_input=[data_object_b["id"]],
+            was_informed_by=nucleotide_sequencing_b_id,
+            id="nmdc:wfmgan-00-000002.1",
+            has_output=[data_object_b["id"]],
+        )[0]
+
+        data_generation_set = seeded_db.get_collection(name="data_generation_set")
+        workflow_execution_set = seeded_db.get_collection(name="workflow_execution_set")
+        data_object_set = seeded_db.get_collection(name="data_object_set")
+
+        data_generation_set.insert_many([nucleotide_sequencing_b])
+        workflow_execution_set.insert_many([metagenome_annotation_b])
+        data_object_set.insert_many([data_object_b, data_object_b])
+
+        # Update the `alldocs` collection, which is a cache used by the endpoint under test.
+        ensure_alldocs_collection_has_been_materialized(force_refresh_of_alldocs=True)
+
+        yield seeded_db
+
+        # Clean up: Delete the documents we created within this fixture, from the database.
+        data_generation_set.delete_many({"id": nucleotide_sequencing_b["id"]})
+        workflow_execution_set.delete_many({"id": metagenome_annotation_b["id"]})
+        data_object_set.delete_many(
+            {"id": {"$in": [data_object_b["id"], data_object_b["id"]]}}
```

Comment:
I think there was a copy/paste error here.

```suggestion
            {"id": {"$in": [data_object_b["id"]]}}
```